src/mpeg_decode.c

```diff
@@ -218,6 +218,12 @@ mpeg_dec_fill_sfinfo (mpg123_handle *mh, SF_INFO *info)
 	info->channels = channels ;
 
 	length = mpg123_length (mh) ;
+	if (length == 0 && !psf->is_pipe)
```

Comment:
Shouldn't `length == 0` be `length <= 0` instead?

Reply:
Yes I think so, thanks!